#!/usr/bin/env python3

from unittest import TestCase
# catch potential exception from import
try:
    from public.restaurant import Restaurant
    from public.onsite_restaurant import OnsiteRestaurant
    from public.delivery_restaurant import DeliveryRestaurant
except Exception:
    # Just make sure that all tests are still executed to have a stable number
    # of exercise points. An appropriate warning is generated by the smoke tests.
    pass

class TestInheritance(TestCase):

    def test_onsite_restaurant_extends_restaurant(self):
        try:
            sut = OnsiteRestaurant("osr", "all cuisine", 20)
        except:
            m = "@@OnsiteRestaurant cannot be instantiated.@@"
            self.fail(m)
        self.assertIsInstance(sut, Restaurant, "@@OnsiteRestaurant does not extend Restaurant.@@")

    def test_delivery_restaurant_extends_restaurant(self):
        try:
            sut = DeliveryRestaurant("dr", "all cuisine", 20)
        except:
            m = "@@DeliveryRestaurant cannot be instantiated.@@"
            self.fail(m)
        self.assertIsInstance(sut, Restaurant, "@@DeliveryRestaurant does not extend Restaurant.@@")

